[TASK] Update and improve cache section in DI chapter (#5213)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/DependencyInjection/Index.rst

contain PHP classes and it is the fundamental setup we will outline in the following sections.

For developers, it is important to understand that dealing with Symfony DI is
an *early core bootstrap and cached* thing. The system will fail upon misconfiguration, leading
to unreachable frontend and backend.

.. attention::

   Errors in the DI configuration may block frontend and backend!

   The DI cache does not heal by itself but needs to be cleared manually!

With the container cache entry being a low level early bootstrap thing that is expensive
to calculate when it has to be rebuild, there is a limited list of options to flush
this cache:

*   The container cache entry is *not deleted* when a backend user clicks "Flush all caches"
    in the backend top toolbar if the instance is configured as :ref:`production <Environment-context>`
    application. For developer convenience, the container cache *is* flushed in development
    context, though.

*   The container cache *is* flushed using "Admin tools" -> "Maintenance" -> "Flush Caches"
    of the Install Tool.

> *   The container cache *is* flushed using the CLI command :shell:`typo3 cache:flush`. Using

Reviewer: ```suggestion
*   The container cache *is* flushed using the CLI command :shell:`vendor/bin/typo3 cache:flush`. Using
```

>     :shell:`typo3/cache:warmup` afterwards will rebuild and cache the container.

Reviewer: ```suggestion
    :shell:`vendor/bin/typo3 cache:warmup` afterwards will rebuild and cache the container.
```

*   The container cache is automatically flushed when using the Extension Manager to load

>     or unload extensions in (non-composer) classic mode.

Reviewer: ```suggestion
    or unload extensions in (non-Composer) classic mode.
```

*   Another way to quickly drop this cache during development is to remove all
    :file:`var/cache/code/di/*` files, which reside in :file:`typo3temp/` in classic mode
    instances or elsewhere in composer mode instances (see :ref:`Environment`).

The main takeaway is: When a developer fiddles with container configuration,
the cache needs to be manually cleared. And if some configuration issue slipped in,
which made the container or DI calculation fail, the system does *not* heal itself and

> needs both a fix of the DI configuration plus probably a cache removal. The standalone Install

Reviewer: ```suggestion
needs both a fix of the `Dependency injection <https://docs.typo3.org/permalink/t3coreapi:dependency-injection>`_ configuration plus probably a cache removal. The standalone Install
```
